Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_cp.nut

```diff
@@ -30,11 +39,97 @@ void function GamemodeCP_Init()
 	AddCallback_EntitiesDidLoad( SpawnHardpoints )
 	AddCallback_GameStateEnter( eGameState.Playing, StartHardpointThink )
+	AddCallback_OnClientConnected(GamemodeCP_InitPlayer)
+	AddCallback_OnClientDisconnected(GamemodeCP_RemovePlayer)
+
+
+	ScoreEvent_SetEarnMeterValues("KillPilot",0.1,0.12)
+	ScoreEvent_SetEarnMeterValues("KillTitan",0,0)
+	ScoreEvent_SetEarnMeterValues("TitanKillTitan",0,0)
+	ScoreEvent_SetEarnMeterValues("PilotBatteryStolen",0,35)
+	ScoreEvent_SetEarnMeterValues("Headshot",0,0.02)
+	ScoreEvent_SetEarnMeterValues("FirstStrike",0,0.05)
+
+	ScoreEvent_SetEarnMeterValues("ControlPointCapture",0.1,0.1)
+	ScoreEvent_SetEarnMeterValues("ControlPointHold",0.02,0.02)
+	ScoreEvent_SetEarnMeterValues("ControlPointAmped",0.2,0.15)
+	ScoreEvent_SetEarnMeterValues("ControlPointAmpedHold",0.05,0.05)
+
+	ScoreEvent_SetEarnMeterValues("HardpointAssault",0.10,0.15)
+	ScoreEvent_SetEarnMeterValues("HardpointDefense",0.5,0.10)
+	ScoreEvent_SetEarnMeterValues("HardpointPerimeterDefense",0.1,0.12)
+	ScoreEvent_SetEarnMeterValues("HardpointSiege",0.1,0.15)
+	ScoreEvent_SetEarnMeterValues("HardpointSnipe",0.1,0.15)
+
 }
+void function GamemodeCP_OnPlayerKilled(entity victim, entity attacker, var damageInfo)
+{
+	HardpointStruct attackerCP
+	HardpointStruct victimCP
+
+	if(!attacker.IsPlayer())
+		return
+
+	foreach(HardpointStruct hardpoint in file.hardpoints)
+	{
+		if(hardpoint.imcCappers.contains(victim))
+			victimCP = hardpoint
+		if(hardpoint.militiaCappers.contains(victim))
+			victimCP = hardpoint
+
+		if(hardpoint.imcCappers.contains(attacker))
+			attackerCP = hardpoint
+		if(hardpoint.militiaCappers.contains(attacker))
+			attackerCP = hardpoint
+	}
+
+	if((victimCP.hardpoint!=null)&&(attackerCP.hardpoint!=null))
+	{
+		if(victimCP==attackerCP)
+		{
+			if(victimCP.hardpoint.GetTeam()==attacker.GetTeam())
+			{
+				AddPlayerScore( attacker , "HardpointDefense", victim )
+				attacker.AddToPlayerGameStat(PGS_DEFENSE_SCORE,POINTVALUE_HARDPOINT_DEFENSE)
+			}
+			else if((victimCP.hardpoint.GetTeam()==victim.GetTeam())||(GetHardpointCappingTeam(victimCP)==victim.GetTeam()))
+			{
+				AddPlayerScore( attacker, "HardpointAssault", victim )
+				attacker.AddToPlayerGameStat(PGS_ASSAULT_SCORE,POINTVALUE_HARDPOINT_ASSAULT)
+			}
+		}
+	}
+	else if((victimCP.hardpoint!=null))//siege or snipe
+	{
+
+		if(Distance(victim.GetOrigin(),attacker.GetOrigin())>=1875)//1875 inches(units) are 47.625 meters
+		{
+			AddPlayerScore( attacker , "HardpointSnipe", victim )
+			attacker.AddToPlayerGameStat(PGS_ASSAULT_SCORE,POINTVALUE_HARDPOINT_SNIPE)
+		}
+		else{
+			AddPlayerScore( attacker , "HardpointSiege", victim )
+			attacker.AddToPlayerGameStat(PGS_ASSAULT_SCORE,POINTVALUE_HARDPOINT_SIEGE)
+		}
+	}
+	else if(attackerCP.hardpoint!=null)//Perimeter Defense
+	{
+		if(attackerCP.hardpoint.GetTeam()==attacker.GetTeam())
+			AddPlayerScore( attacker , "HardpointPerimeterDefense", victim)
+			attacker.AddToPlayerGameStat(PGS_DEFENSE_SCORE,50)
```

Comment:
there's a const for that https://github.com/Syampuuh/Titanfall2/blob/9bfcea43ecf2a12fc3783ce22e63e439f422c045/scripts/vscripts/_settings.nut#L172